upload/src/addons/SV/WarningImprovements/XF/Entity/User.php

```diff
@@ -10,9 +10,82 @@
  */
 class User extends XFCP_User
 {
-    public function canViewWarnings()
+    public function canViewWarningActions(&$error = null, User $asUser = null)
```

Comment:
FYI,the recommended practice is for the caller to use rather than an $asUser parameter;
```php
$canViewWarningActions = \XF:asVisitor($user, function() use ($entity ){
    return $entity->canViewWarningActions();
})
```

Reply:
Don't think I really needed to make use of asVisitor because permission isn't being checked in foreach loop just yet so....

Reply:
k, I expect we'll be able to remove it tho